asynction/security.py

```diff
@@ -0,0 +1,401 @@
+    scope_validate_func = None
+    if scheme.x_scope_validate_func:
+        scope_validate_func = load_handler(scheme.x_scope_validate_func)
+
+    if not scope_validate_func:
+        scope_validate_func = validate_scopes
+
+    return scope_validate_func
+
+
+def load_basic_info_func(scheme: SecurityScheme) -> BasicInfoFunc:
+    if scheme.x_basic_info_func is not None:
+        basic_info_func = load_handler(scheme.x_basic_info_func)
+        if not basic_info_func:
+            raise SecurityException("Missing basic info func")
+        return basic_info_func
+    else:
+        raise SecurityException("Missing basic info func")
+
+
+def load_token_info_func(scheme: SecurityScheme) -> TokenInfoFunc:
+    if scheme.x_token_info_func is not None:
+        token_info_func = load_handler(scheme.x_token_info_func)
+        if not token_info_func:
+            raise SecurityException("Missing token info function")
+        return token_info_func
+    else:
+        raise SecurityException("Missing token info function")
+
+
+def load_api_key_info_func(scheme: SecurityScheme) -> APIKeyInfoFunc:
+    if scheme.x_api_key_info_func is not None:
+        token_info_func = load_handler(scheme.x_api_key_info_func)
+        if not token_info_func:
+            raise SecurityException("Missing API Key info function")
+        return token_info_func
+    else:
+        raise SecurityException("Missing API Key info function")
+
+
+def build_http_security_check(
+    requirement: InternalSecurityRequirement, scheme: SecurityScheme
+) -> Optional[InternalSecurityCheck]:
+    _, required_scopes = requirement
+    if scheme.scheme == HTTPAuthenticationScheme.BASIC:
+        basic_info_func = load_basic_info_func(scheme)
+        scope_validate_func = load_scope_validate_func(scheme)
+
+        def http_security_check(request: Request) -> InternalSecurityCheckResponse:
+            token_info = validate_basic(request, basic_info_func, required_scopes)
+            if token_info is None:
+                return None, None
+
+            token_scopes = token_info.get("scope", token_info.get("scopes", ""))
+
+            if not scope_validate_func(required_scopes, token_scopes):
+                return (
+                    None,
+                    f"Invalid scopes: required: {required_scopes}, provided: {token_scopes}",  # noqa: 501
+                )
+
+            return token_info, None
+
+        return http_security_check
+    elif scheme.scheme == HTTPAuthenticationScheme.BEARER:
+        api_key_info_func = load_api_key_info_func(scheme)
+        scope_validate_func = load_scope_validate_func(scheme)
+
+        bearer_format = scheme.bearer_format
+
+        def http_bearer_security_check(
+            request: Request,
+        ) -> InternalSecurityCheckResponse:
+            token_info = validate_api_key(
+                request, api_key_info_func, required_scopes, bearer_format
```

Comment:
I see connexion supports a separate function (`x-bearerInfoFunc`) for the Bearer scheme. Why would we want to use the same one as the API key scheme?

Reply:
Oh I guess because we pass an extra `bearer_format` argument to the `api_key_info_func` (which also answers the other comment I had with regards to the extra argument).

Should we stay consistent with connexion's design decisions and have to distinct methods?